module/Activity/src/Activity/Controller/ActivityController.php

```diff
@@ -63,13 +63,18 @@ public function createAction()
                 $data['costs'] = '-1';    // Hack. Because empty string is seen as 0
             }
 
-            $form->setData($this->getRequest()->getPost());
+            $postData = $this->getRequest()->getPost();
+            $form->setData($postData);
 
             if ($form->isValid()) {
```

Comment:
I think it would be better to validate the form in `createActivity` instead, since it simplifies the logic here and makes the `createActivity` method more robust.